pkg/cmd/roachtest/sqlalchemy.go

```diff
@@ -0,0 +1,382 @@
+				letter = strings.ToLower(letter + letter)
+			}
+			if _, err := db.Exec(fmt.Sprintf("create database db%s", letter)); err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		t.Status("cloning SQL Alchemy and installing prerequisites")
+
+		if err := c.RunE(ctx, c.All(), `sudo apt-get -qq update`); err != nil {
+			t.Fatal(err)
+		}
+		if err := c.RunE(
+			ctx, c.All(), `sudo apt-get -qq install make python3 libpq-dev python-dev gcc`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := c.RunE(
+			ctx, c.All(), `curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := c.RunE(
+			ctx, c.All(), `sudo python3 get-pip.py`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := c.RunE(
+			ctx, c.All(), `sudo pip install pytest -q`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := c.RunE(
+			ctx, c.All(), `sudo pip install psycopg2-binary -q`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := c.RunE(ctx, c.All(), `rm -rf /mnt/data1/sqlalchemy`); err != nil {
+			t.Fatal(err)
+		}
+		if err := c.GitCloneE(
+			ctx,
+			"https://github.com/sqlalchemy/sqlalchemy.git",
+			"/mnt/data1/sqlalchemy",
+			"rel_1_2_18",
+			c.All(),
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		// SQLAlchemy does a version check in a weird way. Override that check.
+		if err := c.RunE(
+			ctx, c.All(),
+			`sed -i 's/`+
+				`        v = connection.execute("select version()").scalar()/`+
+				`        v = connection.execute("select version()").scalar()\n`+
+				`        if v.startswith("CockroachDB"):\n`+
+				`            return 9, 5, None/'`+
+				` /mnt/data1/sqlalchemy/lib/sqlalchemy/dialects/postgresql/base.py`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		// Check the version of Cockroach and find the correct blacklist.
+		version, err := fetchCockroachVersion(ctx, c, c.Node(1)[0])
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Prepend the version with the test name so we can have different
+		// blacklists for each.
+		prependedVersion := fmt.Sprintf("%s %s", t.Name(), version)
+		blacklistName, expectedFailureList, _, _ := sqlalchemyBlacklists.getLists(prependedVersion)
+		if expectedFailureList == nil {
+			t.Fatalf("No SQL Alchemy blacklist defined for cockroach version %s with test set %s -- %s",
+				version, t.Name(), prependedVersion,
+			)
+		}
+		c.l.Printf("Running cockroach version %s, using blacklist %s", version, blacklistName)
+
+		t.Status("running SQL Alchemy test suite")
+		var excludedTests string
+		for _, badTest := range knownTestCrashes {
+			excludedTests += " and not " + badTest
+		}
+
+		runTestCommand := func(testset string) string {
+			dbName := testset
+			if unicode.IsUpper(rune(testset[0])) {
+				dbName = strings.ToLower(dbName + dbName)
+			}
+			return fmt.Sprintf("cd /mnt/data1/sqlalchemy/ && "+
+				"py.test --dburi=\"postgresql://root@localhost:26257/db%s?sslmode=disable\" "+
+				"--maxfail=10000 "+
+				"-k \"test_%s%s\" ", dbName, testset, excludedTests)
```

Comment:
As I said before this blind pattern match is not suitable because it makes the same test run multiple times.

My recommendation is to dig into the source code of `py.test` and find how the `-k` parameter is handled. 

1) if it's a regexp, you know what to do
2) if it's not, we need to patch it for the purpose of this test.

I can help with the patching once you figure out where about the argument is handled in the source code.